File: CH 3. Greedy/3-3.js
// 3-3. 숫자 카드 게임


const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});



var input = [];
var n = 0;

rl.on("line", (line) => {
    n = n + 1;
    input.push(line.split(" ").map(x => Number(x)));

    if (n === input[0][0] + 1) {
        rl.close();
    }
});

rl.on("close", () => {
    var n = input[0][0];
    var m = input[0][1];

    var cards = []

    for (let i = 0; i < n; i++) {
        cards.push(input[i + 1]);
    }


    function solution(cards) {
        var answer = cards[0];

        for (let i = 0; i < cards.length; i++) {
            if (Math.min.apply(null, cards[i]) > Math.min.apply(null, answer)) {
                answer = cards[i];
            }
        }

        return Math.min.apply(null, answer);
    }


    console.log(solution(cards));

    process.exit();
})

File: This Is Coding Test/CH 3. Greedy/3-3.py
# 3-3. 숫자 카드 게임


def solution(cards):
    answer = cards[0]

    for cur in cards:
        if min(cur) > min(answer):
            answer = cur

    return min(answer)


n, m = map(int, input().split())

cards = [list(map(int, input().split())) for _ in range(0, n)]


print(solution(cards))
